lib/proof.rb

```diff
@@ -0,0 +1,55 @@
+class Proof < FakeRecord
+  class ProversValidator < ActiveModel::EachValidator
+    def validate_each(record, attribute, value)
+      not_provers = value.reject { |id| Prover.where(id: id).any? }
+      if not_provers.any?
+        record.errors.add attribute, "#{not_provers} are not provers"
+      end
+    end
+  end
+
+  attr_reader :proof_obligation, :provers, :ontology
+
+  validates :provers, provers: true
+
+  def initialize(opts)
+    opts[:proof] ||= {}
+    opts[:proof][:provers] ||= []
+
+    @ontology = Ontology.find(opts[:ontology_id])
+    # HACK: remove the empty string from params
+    # Rails 4.2 introduces the html form option :include_hidden
+    @provers = opts[:proof][:provers].select(&:present?).map(&:to_i)
```

Comment:
doesn't belong a hack massaging the params to the controller or a helper called from the controller?